src/run.rs

```diff
@@ -255,7 +258,8 @@ impl Handler for ExecHandler {
 }
 
 pub fn run(args: Args) -> Result<()> {
-    watch::<ExecHandler>(args)
+    let handler = ExecHandler::new(&args)?;
+    watch(&handler)
```

Comment:
Removing mutability from the `Handler` trait methods enables us to both initialize `ExecHandler` from a `&Arg` and pass a `&Handler` to `watch` here; if the mutability constraint were left on the trait methods than we would have to clone one or both here in `run` which while it isn't the end of the world, feels kind of gross to me. Since I didn't see that mutability being used anywhere it seemed safe to remove.

Reply:
The mutability was mostly there for maximum compatibility, given it's a user-implemented trait. If the user wants to mutate things, then they can. However, that can also be done using interior mutability so it's worth the loss.